File: src/main/java/com/mpe85/grampa/event/PostParseEvent.java
package com.mpe85.grampa.event;

import com.mpe85.grampa.runner.ParseResult;

/**
 * Event posted after a parse run.
 * 
 * @author mpe85
 *
 * @param <T>
 *            the type of the stack elements
 */
public class PostParseEvent<T> {
	
	public PostParseEvent(final ParseResult<T> result) {
		this.result = result;
	}
	
	public ParseResult<T> getResult() {
		return result;
	}
	
	private final ParseResult<T> result;
	
}

File: src/main/kotlin/com/mpe85/grampa/event/PostParseEvent.kt
package com.mpe85.grampa.event

import com.mpe85.grampa.parser.ParseResult

/**
 * Event posted after a parse run.
 *
 * @author mpe85
 * @param[T] The type of the stack elements
 * @property[result] The parse result
 */
public class PostParseEvent<T>(public val result: ParseResult<T>)
